Mathlib/FieldTheory/Differential/Liouville.lean

```diff
@@ -0,0 +1,175 @@
+    (inst1 : IsLiouville F K) (inst2 : IsLiouville K A) : IsLiouville F A where
+  is_liouville (a : F) (ι : Type) [Fintype ι] (c : ι → F) (hc : ∀ x, (c x)′ = 0)
+      (u : ι → A) (v : A) (h : a = ∑ x, c x * logDeriv (u x) + v′) := by
+    obtain ⟨ι'', _, c'', hc', u'', v', h'⟩ := inst2.is_liouville (a : K) ι
+        ((↑) ∘ c)
+        (fun _ ↦ by simp only [Function.comp_apply, ← coe_deriv, lift_map_eq_zero_iff, hc])
+        ((↑) ∘ u) v (by simpa only [Function.comp_apply, ← IsScalarTower.algebraMap_apply])
+    have hc (x : ι'') := mem_range_of_deriv_eq_zero F (hc' x)
+    choose c' hc using hc
+    apply inst1.is_liouville a ι'' c' _ u'' v'
+    · rw [h']
+      simp [hc]
+    · intro
+      apply_fun ((↑) : F → K)
+      · simp only [Function.comp_apply, coe_deriv, hc, algebraMap.coe_zero]
+        apply hc'
+      · apply NoZeroSMulDivisors.algebraMap_injective
+
+section Algebraic
+/-
+The case of Liouville's theorem for algebraic extension.
+-/
+
+variable {F K} [CharZero F]
+
+instance (B : IntermediateField F K)
+    [FiniteDimensional F B] [inst : IsLiouville F K] :
+    IsLiouville F B where
+  is_liouville (a : F) (ι : Type) [Fintype ι] (c : ι → F) (hc : ∀ x, (c x)′ = 0)
+      (u : ι → B) (v : B) (h : a = ∑ x, c x * logDeriv (u x) + v′) := by
+    apply inst.is_liouville a ι c hc (B.val ∘ u) (B.val v)
+    dsimp only [coe_val, Function.comp_apply]
+    conv =>
+      rhs
+      congr
+      · rhs
+        intro x
+        rhs
+        apply logDeriv_algebraMap (u x)
+      · apply (deriv_algebraMap v)
+    simp_rw [IsScalarTower.algebraMap_apply F B K]
+    norm_cast
+
+
+lemma IsLiouville.equiv {K' : Type*} [Field K'] [Differential K'] [Algebra F K']
+    [DifferentialAlgebra F K'] [Algebra.IsAlgebraic F K']
+    [inst : IsLiouville F K] (e : K ≃ₐ[F] K') : IsLiouville F K' where
+  is_liouville (a : F) (ι : Type) [Fintype ι] (c : ι → F) (hc : ∀ x, (c x)′ = 0)
+      (u : ι → K') (v : K') (h : a = ∑ x, c x * logDeriv (u x) + v′) := by
+    apply inst.is_liouville a ι c hc (e.symm ∘ u) (e.symm v)
+    apply_fun e.symm at h
+    simpa [AlgEquiv.commutes, map_add, map_sum, map_mul, logDeriv, algEquiv_deriv'] using h
+
+private local instance liouville_of_finiteDimensional_galois [FiniteDimensional F K]
+    [IsGalois F K] : IsLiouville F K where
+  is_liouville (a : F) (ι : Type) [Fintype ι] (c : ι → F) (hc : ∀ x, (c x)′ = 0)
+      (u : ι → K) (v : K) (h : a = ∑ x, c x * logDeriv (u x) + v′) := by
+    haveI : CharZero K := charZero_of_injective_algebraMap
+      (NoZeroSMulDivisors.algebraMap_injective F K)
+    let c' (i : ι) := (c i) / (Fintype.card (K ≃ₐ[F] K))
+    let u'' (i : ι) := ∏ x : (K ≃ₐ[F] K), x (u i)
+    have : ∀ i, u'' i ∈ fixedField (⊤ : Subgroup (K ≃ₐ[F] K)) := by
+      rintro i ⟨e, _⟩
+      change e (u'' i) = u'' i
+      simp only [u'', map_prod]
+      apply Fintype.prod_equiv (Equiv.mulLeft e)
+      simp
+    have ffb := (IsGalois.tfae.out 0 1).mp (inferInstanceAs (IsGalois F K))
```

Comment:
It is probably safer to make explicit the type of `ffb` in case somebody, later, refactors `IsGalois.tfae`.